src/teehr/visualization/dataframe_accessor.py

```diff
@@ -0,0 +1,176 @@
+"""Provides the teehr accessor extending pandas DataFrames."""
+import itertools
+from math import pi
+import random
+import pandas as pd
+import logging
+from pathlib import Path
+
+from bokeh.plotting import figure, save, output_file, show
+from bokeh.palettes import Turbo256
+
+logger = logging.getLogger(__name__)
+
+
+@pd.api.extensions.register_dataframe_accessor("teehr")
+class TEEHRDataFrameAccessor:
+    """Extends pandas DataFrame objects.
+
+    Notes
+    -----
+    This class contains example methods for summarizing and plotting metrics
+    as well as timeseries. This requires more validation in each method to
+    ensure the DataFrame has the appropriate data.
+
+    Methods operating on metrics data should start with 'metrics_' and methods
+    operating on timeseries data should start with 'timeseries_'.
+    """
+
+    def __init__(self, pandas_obj):
+        """Initialize the class."""
+        self._df = pandas_obj
+        self._validate(pandas_obj)
+
+    @staticmethod
+    def _validate(obj):
+        """Validate the DataFrame object columns."""
+        if "location_id" not in obj.columns:
+            raise AttributeError("Must have 'location_id'.")
+        if obj.index.size == 0:
+            raise AttributeError("DataFrame must have data.")
+
+    def _timeseries_unique_values(
+            self,
+            variable_df: pd.DataFrame,
+    ) -> dict:
+        """Get dictionary of all unique values of each column."""
+        logger.info("Retrieving unique values from DataFrame.")
+        columns = variable_df.columns.to_list()
+        Dict = {}
```

Comment:
A bit nitpicky, but maybe `Dict` should be something more descriptive? This could cause conflict if the `typing` library is used, ie., `from typing import Dict`